.github/workflows/build-and-deploy.yml

```diff
@@ -1,166 +1,163 @@
-      run: bash .github/workflow-scripts/verify_deployment.sh ${{ secrets.STAGING_SEPOLIA_URL }} ${{ needs.docker_build_and_publish.outputs.IMAGE_TAG }}
+    steps:
+      - name: Setup JFrog CLI
+        uses: jfrog/setup-jfrog-cli@v4
+        env:
+          JF_URL: ${{ vars.JFROG_URL}}
+          JF_ACCESS_TOKEN: ${{ secrets.ARTIFACTORY_ANGKOR_CONTRIBUTOR }}
+
+      - name: Promote to Staging
+        run: |
+          jf rt dpr juno/${{ env.DOCKER_IMAGE_TAG }} ${{ env.REPO_DEV }} ${{ env.REPO_STAGING }}
 
   staging-starknet-rs-tests:
-    needs: [deploy_to_staging]
+    needs: [promote_to_staging]
     uses: ./.github/workflows/starknet-rs-tests.yml
     secrets:
       STARKNET_RPC: ${{ secrets.STAGING_SEPOLIA_URL }}/v0_6
-      
+
   staging-starknet-js-tests:
-    needs: [deploy_to_staging]
+    needs: [promote_to_staging]
     uses: ./.github/workflows/starknet-js-tests.yml
     secrets:
       TEST_RPC_URL: ${{ secrets.STAGING_SEPOLIA_URL }}/v0_7
       TEST_ACCOUNT_ADDRESS: ${{ secrets.TEST_ACCOUNT_ADDRESS }}
       TEST_ACCOUNT_PRIVATE_KEY: ${{ secrets.TEST_ACCOUNT_PRIVATE_KEY }}
 
-  deploy_to_production:
-    needs: [docker_build_and_publish, deploy_to_staging]
+  promote_to_production:
+    needs: [build_docker_image, promote_to_staging]
     runs-on: ubuntu-latest
     environment:
       name: Production
     steps:
-      - name: Repository Dispatch Prod
-        env: 
-          EVENT_NAME: juno-prod
-          IMAGE_TAG: ${{ needs.docker_build_and_publish.outputs.IMAGE_TAG }}
-          MAINNET: apps/juno-prod/overlays/prod-mainnet/config.yaml
-          SEPOLIA: apps/juno-prod/overlays/prod-sepolia/config.yaml
-          SEPOLIA_INTEGRATION: apps/juno-prod/overlays/prod-sepolia-integration/config.yaml
+      - name: Setup JFrog CLI
+        uses: jfrog/setup-jfrog-cli@v4
+        env:
+          JF_URL: ${{ vars.JFROG_URL}}
+          JF_ACCESS_TOKEN: ${{ secrets.ARTIFACTORY_ANGKOR_CONTRIBUTOR }}
+
+      - name: Promote to Production
         run: |
-          curl -L \
-          -X POST \
-          -H "Accept: application/vnd.github+json" \
-          -H "Authorization: token ${{ secrets.ACCESS_TOKEN }}" \
-          -H "X-GitHub-Api-Version: 2022-11-28" \
-          https://api.github.com/repos/NethermindEth/argo/dispatches \
-          -d '{"event_type": "${{ env.EVENT_NAME }}", "client_payload":{"name": "${{ env.EVENT_NAME }}", "mainnet_config": "${{ env.MAINNET }}", "sepolia_config": "${{ env.SEPOLIA }}", "sepolia_integration_config": "${{ env.SEPOLIA_INTEGRATION }}", "tag": "${{ env.IMAGE_TAG }}"}}'
-          
+          jf rt dpr juno/${{ env.DOCKER_IMAGE_TAG }} ${{ env.REPO_STAGING }} ${{ env.REPO_PROD }}
+
+      - name: Authenticate App
+        id: gh-app
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ vars.APP_ID }}
+          private-key: ${{ secrets.APP_PRIVATE_KEY }}
+
+      - name: Trigger juno prod deployment workflow
```

Comment:
Do we need this step? I guess we should do same here as for staging like:

```
 - name: Promote to Prod
        run: |
          jf rt dpr juno/${{ env.DOCKER_IMAGE_TAG }} ${{ env.REPO_DEV }} ${{ env.REPO_PROD }}
```

Reply:
Oh, I see it's already aboved, so I guess we can just remove the old part

Reply:
that's correct, I forgot to remove that when I was cherrypicking and rebasing, let me fix